plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/price/FlexibleEnginePriceCalculator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.plugins.flexibleengine.price;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.models.billing.Price;
+import org.eclipse.xpanse.modules.models.billing.ResourceUsage;
+import org.eclipse.xpanse.modules.models.billing.ServicePrice;
+import org.eclipse.xpanse.modules.models.billing.enums.BillingMode;
+import org.eclipse.xpanse.modules.models.billing.enums.PricingPeriod;
+import org.eclipse.xpanse.modules.models.billing.exceptions.ServicePriceCalculationFailed;
+import org.eclipse.xpanse.modules.orchestrator.price.ServicePriceRequest;
+import org.springframework.stereotype.Component;
+
+/**
+ * Class that implements the price calculation for FlexibleEngine.
+ */
+@Slf4j
+@Component
+public class FlexibleEnginePriceCalculator {
+
+
+    /**
+     * Get the price of the service.
+     *
+     * @param request service price request
+     * @return price
+     */
+    public ServicePrice getServicePrice(ServicePriceRequest request) {
+
+        if (request.getBillingMode() == BillingMode.PAY_PER_USE) {
+            return getServicePriceWithPayPerUse(request);
+        } else {
+            return getServicePriceWithFixed(request);
+        }
+    }
+
+    private ServicePrice getServicePriceWithPayPerUse(ServicePriceRequest request) {
+        ResourceUsage resourceUsage = request.getFlavor().getPricing().getResourceUsage();
+        if (Objects.isNull(resourceUsage)) {
+            String errorMsg = String.format("ResourceUsage for flavor %s is null.",
+                    request.getFlavor().getName());
+            throw new ServicePriceCalculationFailed(errorMsg);
+        }
+        ServicePrice servicePrice = new ServicePrice();
+        servicePrice.setPeriodCount(1);
+        boolean markPriceNotNull = Objects.nonNull(resourceUsage.getMarkUpPrice());
```

Comment:
What do `license` and `markup` mean respectively, and what are the corresponding variables in Huawei Cloud? Not 'amount' and `officialAmount` ?

Reply:
these two values are not part of the Huaweicloud API. 
license fees - fee what the ISV charges additionally for the softwares used in the managed service. This can be one-time or recurring. 
Markup fee - This means, the extra charge the service vendor always adds on top of the resource usage. This is the kind of "service charge" the ISV gets. Again this can be one-time or recurring.

Reply:
Understood and updated code.